src/frontend/src/pages/ChangeRequestDetailPage/DeleteChangeRequestView.tsx

```diff
@@ -0,0 +1,103 @@
+import { DeleteChangeRequestInputs } from './DeleteChangeRequest';
+
+interface DeleteChangeRequestViewProps {
+  changeRequest: ChangeRequest;
+  modalShow: boolean;
+  onHide: () => void;
+  onSubmit: (data: DeleteChangeRequestInputs) => Promise<void>;
+}
+
+const DeleteChangeRequestView: React.FC<DeleteChangeRequestViewProps> = ({ changeRequest, modalShow, onHide, onSubmit }) => {
+  const changeRequestIdTester = (crId: string | undefined) => crId !== undefined && crId === changeRequest.crId.toString();
+
+  const schema = yup.object().shape({
+    crId: yup.string().required().test('cr-id-test', 'Change Request ID does not match', changeRequestIdTester)
+  });
+
+  const {
+    handleSubmit,
+    control,
+    formState: { errors }
+  } = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      crId: ''
+    }
+  });
+
+  const onSubmitWrapper = async (data: DeleteChangeRequestInputs) => {
+    await onSubmit(data);
+  };
+
+  return (
+    <Dialog fullWidth maxWidth="md" open={modalShow} onClose={onHide}>
+      <DialogTitle className={'font-weight-bold'}>{`Delete Change Request #${changeRequest.crId}`}</DialogTitle>
+      <DialogContent>
+        <Typography>Are you sure you want to delete Change Request #{changeRequest.crId}?</Typography>
+        <Typography sx={{ fontWeight: 'bold' }}>This action cannot be undone!</Typography>
+        <form
+          id="delete-cr-form"
+          onSubmit={(e) => {
+            e.preventDefault();
+            e.stopPropagation();
+            handleSubmit(onSubmitWrapper)(e);
+          }}
+        >
+          <Controller
+            name="crId"
+            control={control}
+            rules={{ required: true }}
+            render={({ field: { onChange, value } }) => (
+              <>
+                <Typography
+                  sx={{
+                    paddingTop: 1,
+                    paddingBottom: 1
+                  }}
+                >
+                  To confirm deletion, please type in the ID number of this Change Request.
+                </Typography>
+                <TextField
+                  required
+                  variant="outlined"
+                  id="crId-input"
+                  type="number"
+                  autoComplete="off"
+                  onChange={onChange}
+                  value={value}
+                  fullWidth
+                  sx={{ width: 1 }}
+                  error={!!errors.crId}
+                  placeholder="Enter Change Request ID here"
+                  helperText={errors.crId?.message}
+                />
+              </>
+            )}
+          />
+        </form>
+      </DialogContent>
+      <DialogActions>
+        <NERSuccessButton variant="contained" sx={{ mx: 1 }} onClick={onHide}>
+          Cancel
+        </NERSuccessButton>
+        <NERFailButton variant="contained" type="submit" form="delete-cr-form" sx={{ mx: 1 }}>
+          Delete
+        </NERFailButton>
```

Comment:
We should ask james ab the colors for these, a bit weird since success is a delete